service/controller/clusterapi/v28/key/machine_deployment.go

```diff
@@ -1,26 +1,72 @@
 package key
 
 import (
+	"math/bits"
+	"net"
 	"strconv"
 
 	g8sv1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/ipam"
 	"github.com/giantswarm/microerror"
 	"sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
 
 	"github.com/giantswarm/aws-operator/pkg/label"
 )
 
+func WorkerSubnet(cr v1alpha1.MachineDeployment) string {
+	return cr.Labels[label.MachineDeploymentSubnet]
+}
+
 func WorkerClusterID(cr v1alpha1.MachineDeployment) string {
 	return cr.Labels[label.Cluster]
 }
 
 // TODO this method has to be properly implemented and renamed eventually.
 func StatusAvailabilityZones(cr v1alpha1.MachineDeployment) []g8sv1alpha1.AWSConfigStatusAWSAvailabilityZone {
+	var workerSubnet net.IPNet
+	{
+		s := WorkerSubnet(cr)
+		if s == "" {
+			//return nil, microerror.Maskf(notFoundError, "MachineDeployment is missing subnet allocation")
+			panic("MachineDeployment is missing subnet allocation")
+		}
+
+		_, n, err := net.ParseCIDR(s)
+		if err != nil {
+			//return nil, microerror.Mask(err)
+			panic(err)
+		}
+
+		workerSubnet = *n
+	}
+
 	var azs []g8sv1alpha1.AWSConfigStatusAWSAvailabilityZone
-	for _, s := range WorkerAvailabilityZones(cr) {
-		azs = append(azs, g8sv1alpha1.AWSConfigStatusAWSAvailabilityZone{
-			Name: s,
-		})
+	{
+		azsSubnets, err := splitNetwork(workerSubnet, uint(len(WorkerAvailabilityZones(cr))))
```

Comment:
This looks like the annotation in the MachineDeployment holds the whole cluster network and not the subnet specific to a particular MachineDeployment. Which value is the annotation holding now? I am confused.

Reply:
At this point when we want to get the cluster up with in-build workers, it holds whole cluster subnet. When we start creating the node pool CF stack, part of this code can be dropped as it's the worker subnet that only needs to be split per AZ.